src/measure_theory/interval_integral.lean

```diff
@@ -394,6 +394,12 @@ begin
     preimage_neg_Ioc, neg_neg, restrict_congr_set Ico_ae_eq_Ioc]
 end
 
+/-- If two functions are equal in the relevant interval, their interval integrals are also equal. -/
+lemma integral_congr {a b : ℝ} (f g : ℝ → E) (h : eq_on f g (interval a b)) :
```

Comment:
May I suggest that we make `f` and `g` implicit here? I think that will make this lemma more user-friendly.

Reply:
I think this is a reasonable suggestion but not clear-cut (see [my previous comment](https://github.com/leanprover-community/mathlib/pull/5724#discussion_r559053709)).  Up to @mmasdeu!

Reply:
I think, we should have this lemma for any measure, not only `volume`. And I would prefer to have the same `{f g}` vs `(f g)` decision in this lemma, its version for set integrals, and its version for integrals over the whole space.

Reply:
@mmasdeu To address @urkud 's suggestion, this works (with the same proof):
```
lemma integral_congr {f g : α → E} (h : eq_on f g (interval a b)) :
  ∫ x in a..b, f x ∂μ = ∫ x in a..b, g x ∂μ :=
```
Note that I'm generalizing here from `ℝ` to a general type `α` which is a linear order (and a few other hypotheses).  To make this work, the lemma needs to be moved down ~10 lines, the point at which `variables [order_closed_topology α]` is introduced (to make `is_measurable_Ioc` work).

Reply:
Thank you all! I made the arguments implicit (seems to be the right thing to do) and moved as per @hrmacbeth's suggestion.